treeherder/auth/backends.py

```diff
@@ -49,12 +49,23 @@ def _get_email(self, result):
         For more info on scopes:
         https://docs.taskcluster.net/manual/3rdparty#authenticating-with-scopes
         """
+        # Try finding the email in the mozilla-user scope
         email = self._get_scope_value(result["scopes"], "assume:mozilla-user:")
 
         if email and re.search(r'.+@.+', email):
             return email
+        else:
+            # Try finding the email in the clientId.
+            # Credit for regex to http://emailregex.com/ Python section
+            match = re.search(
+                r"([a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+)",
```

Comment:
Should we explicitly trim off the `email/` prefix rather than just extracting any email-like string? It just seems like this regex would extract an email from eg `fake/joe@smith.com/fake` which may or may not be covered by the protections that Dustin mentioned in the Vidyo call.

@djmitche, does this look ok to you? Or would extracting the email from `auth:create-client:email/foo@mozilla.com/*` be preferred?

Also, I think we said it might be safer to allow using clientID, but to never match against an existing user, so they couldn't inherit sheriff permissions, or someone else's Treeherder Hawk secrets?